swaybar/tray/item.c

```diff
@@ -494,23 +494,33 @@ uint32_t render_sni(cairo_t *cairo, struct swaybar_output *output, double *x,
 	}
 
 	int padded_size = icon_size + 2*padding;
-	*x -= padded_size;
+	double padded_surface_size = (double) padded_size / output->scale;
+	double x_in_pixels = (*x * output->scale) - padded_size;
```

Comment:
Name nit: "in pixels" is a bit vague, as there are logical (surface-local) pixels and physical (buffer-local) pixels.

Maybe use something like `buffer_x` instead?